source/common/upstream/cluster_manager_impl.cc

```diff
@@ -431,6 +431,19 @@ void ClusterManagerImpl::onClusterInit(Cluster& cluster) {
   // We have a situation that clusters will be immediately active, such as static and primary
   // cluster. So we must have this prevention logic here.
   if (cluster_data != warming_clusters_.end()) {
+    // If there is no secret entity, currently supports only TLS Certificate and Validation
+    // Context, when it failed to extract them via SDS, it will fail to change cluster status from
+    // warming to active. In current implementation, there is no strategy to activate clusters
+    // which failed to initialize at once.
+    // TODO(shikugawa): To implement to be available by keeping warming after no-available secret
+    // entity behavior occurred. And remove
+    // `envoy.reloadable_features.cluster_keep_warming_no_secret_entity` runtime feature flag.
+    if (Runtime::runtimeFeatureEnabled(
+            "envoy.reloadable_features.cluster_keep_warming_no_secret_entity") &&
+        !cluster.info()->transportSocketMatcher().resolve(nullptr).factory_.isReady()) {
```

Comment:
It seems strange to go through the matcher here with a null match. Shouldn't we actually be checking if *all* transport sockets are ready?

Reply:
null match is the default transport socket factory. we could check all transport factory but some of them may not be used.

Reply:
IMO we don't know what is used, so we need to check if they are all ready and not go through the matcher?

Reply:
Sure, done.